lib/services/user_login_state.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types/userloginstate"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// UserLoginStateGetter defines an interface for reading user login states.
+type UserLoginStateGetter interface {
```

Comment:
Do we really need the interfaces? Do they need to be preemptively declared here?

See https://github.com/golang/go/wiki/CodeReviewComments#interfaces.

Reply:
I can kick this down the road for now. We may not need it for this one.

Reply:
Nice, that's better. Postpone as much as you can, and if you do need them, consider declaring the interfaces where they are used and making them as small as possible.